src/object/cli_obj.c

```diff
@@ -5277,36 +5308,69 @@ obj_update_shards_get(struct dc_object *obj, daos_obj_fetch_t *args, unsigned in
 {
 	uint8_t		*tgt_bitmap;
 	uint32_t	failure_cnt = 0;
+	int		grp_idx;
+	uint32_t	grp_start;
+	uint32_t	shard_nr = 0;
 	int		i;
-	int		rc;
+	int		rc = 0;
 
-	rc = obj_dkey2grpmemb(obj, obj_auxi->dkey_hash, map_ver, shard, shard_cnt);
-	if (rc || !obj_is_ec(obj))
-		return rc;
+	if (!obj_is_ec(obj))
+		return obj_dkey2grpmemb(obj, obj_auxi->dkey_hash, map_ver, shard, shard_cnt);
 
-	tgt_bitmap = obj_auxi->reasb_req.tgt_bitmap;
-	for (i = 0; i < *shard_cnt; i++) {
-		int shard_idx;
+	grp_idx = obj_dkey2grpidx(obj, obj_auxi->dkey_hash, map_ver);
+	if (grp_idx < 0)
+		return grp_idx;
 
-		if (isclr(tgt_bitmap, i))
-			continue;
```

Comment:
Why remove this check?

Reply:
oh, the check has been moved to L5365.  So for update, let's check if there are still valid shards left for the EC file.